crates/sui-graphql-rpc/src/types/dry_run_result.rs

```diff
@@ -0,0 +1,153 @@
+            .events
+            .data
+            .into_iter()
+            .map(|e| NativeEvent {
+                sender: e.sender,
+                package_id: e.package_id,
+                transaction_module: e.transaction_module,
+                type_: e.type_,
+                contents: e.bcs,
+            })
+            .collect();
+        let effects: NativeTransactionEffects =
+            bcs::from_bytes(&results.raw_effects).map_err(|e| {
+                Error::Internal(format!("Unable to deserialize transaction effects: {e}"))
+            })?;
+        let tx_data: NativeTransactionData = bcs::from_bytes(&results.raw_txn_data)
+            .map_err(|e| Error::Internal(format!("Unable to deserialize transaction data: {e}")))?;
+        Ok(Self {
+            error: results.error,
+            results: Some(execution_results),
+            events,
+            effects,
+            tx_data,
+        })
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, SimpleObject)]
+pub(crate) struct DryRunEffect {
+    /// Changes made to arguments that were mutably borrowed by each command in this transaction.
+    pub mutated_references: Option<Vec<DryRunMutation>>,
+
+    /// Return results of each command in this transaction.
+    pub return_values: Option<Vec<DryRunReturn>>,
+}
+
+impl TryFrom<SuiExecutionResult> for DryRunEffect {
+    type Error = crate::error::Error;
+
+    fn try_from(result: SuiExecutionResult) -> Result<Self, Self::Error> {
+        let mutated_references = result
+            .mutable_reference_outputs
+            .iter()
+            .map(|(argument, bcs, type_)| {
+                Ok(DryRunMutation {
+                    input: (*argument).into(),
+                    type_: MoveType::new(type_.clone().try_into()?),
+                    bcs: bcs.into(),
+                })
+            })
+            .collect::<Result<Vec<_>>>()
+            .map_err(|e| {
+                Error::Internal(format!(
+                    "Failed to parse results returned from dev inspect: {:?}",
+                    e
+                ))
+            })?;
+        let return_values = result
+            .return_values
+            .iter()
+            .map(|(bcs, type_)| {
+                Ok(DryRunReturn {
+                    type_: MoveType::new(type_.clone().try_into()?),
+                    bcs: bcs.into(),
+                })
+            })
+            .collect::<Result<Vec<_>>>()
+            .map_err(|e| {
+                Error::Internal(format!(
+                    "Failed to parse results returned from dev inspect: {:?}",
+                    e
+                ))
+            })?;
+        Ok(Self {
+            mutated_references: Some(mutated_references),
+            return_values: Some(return_values),
+        })
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, SimpleObject)]
+pub(crate) struct DryRunMutation {
+    pub input: TransactionArgument,
+
+    #[graphql(name = "type")]
```

Comment:
I think @wlmyng taught me that you don't need to use this annotation to get rid of a trailing underscore like this -- `async-graphql` does that for you.